src/transformStateWithClones.js

```diff
@@ -7,7 +7,22 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let newState = { ...state };
+  const states = [];
+
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      Object.assign(newState, key.extraData);
```

Comment:
Ensure that `key.extraData` is always an object before using `Object.assign`. If `extraData` can be something other than an object, it might cause unexpected behavior.